### PR TITLE
Bug 1901030 - Enable Prettier

### PR DESCRIPTION
**Pull request by sarahhjchung**
[**Bug 1901030**](https://bugzilla.mozilla.org/show_bug.cgi?id=1901030)

Changes made:
- Added a new step to our CI workflow to check for Prettier warnings
- Installed Prettier and set up npm for it

Note:
- Will wait for review on this set-up before running Prettier on all files

### CODE REVIEW

File: package.json
Comment (dmose) [inline package.json:13]:
```suggestion
    "prettier:check:ci": "npx --no prettier . --check"
```
This will prevent npx from trying to do anything updating.

**Review by dmose — COMMENTED** [review]
This looks good to me. I do want to play around with it some more before giving it final review. I'd suggest adding instructions to set up husky to the beginning of CONTRIBUTING.md, since each develop will want to do that as they get started.

File: CONTRIBUTING.md
Comment (sarahhjchung) [inline CONTRIBUTING.md:23]: This section is the only part I added to CONTRIBUTING, the rest was the pre-commit linting.

File: package.json
Comment (sarahhjchung) [reply package.json:13]: Good point, I think I just used the full thing because I saw it on a some documentation that I was following but I'll change it to just "prettier"

File: CONTRIBUTING.md
Comment (sarahhjchung) [inline CONTRIBUTING.md:7]: All changes here are from prettier

**Comment by sarahhjchung** [comment]
I've removed husky and lint-staged now. I'll wait to run prettier until the new changes are reviewed

File: package.json
Comment (dmose) [inline package.json:12]: Actually, I realize now that we probably _do_ want this to be"prettier:check", because we want a "prettier" script as well to run it without `--check`, probably something like:

`npm run prettier . --`

The -- tells npm to pass any remaining command-line arguments through to prettier.

I'd suggest we just ditch `npx` entirely.

**Review by dmose — COMMENTED** [review]
This is looking good.  After you commit the reformatting changes, you'll want to add a file to tell `git blame/annotate` to ignore that commit.  
https://gist.github.com/kateinoigakukun/b0bc920e587851bfffa98b9e279175f2 has details.

**Comment by dmose** [comment]
I suspect you'll want to do the work where you merge from `main`, do the rewrite, and add the ignore file` and then have me do the final review after landing any other in-flight patches that you have, so that you don't have to deal with formatting-related conflicts. Ping me on Slack when you're ready to do that.

**Comment by dmose** [comment]
This looks good.  https://prettier.io/docs/en/install.html suggests adding a .prettierrc that editors and other tools can key off to know that we're using prettier, so once you do that, I think things should be set here.

**Review by dmose — APPROVED** [review]
Looks good to me.  I've added a couple of settings to the vscode workspace file so that it should Just Work and run things through prettier on save.

r=dmose

**Pull request merged**
